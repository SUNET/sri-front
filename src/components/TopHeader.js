import React from 'react';
import { withTranslation } from 'react-i18next';
import { Nav, Navbar, Form, Dropdown } from 'react-bootstrap';
import { NavLink, Link } from 'react-router-dom';
import { slide as BurgerMenu } from 'react-burger-menu';
import Logout from './Logout';
import AccordionMenuOptions from './AccordionMenuOptions';
import { BrowserView, MobileView } from 'react-device-detect';
import PropTypes from 'prop-types';
import { debounce } from '../utils';
import { MILLISECONDS_TO_WAIT_REQUEST_AUTOCOMPLETE } from '../utils/constants';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';

import formattedServicesData from './service/serviceListData';

import '../style/TopHeader.scss';

export class TopHeader extends React.Component {
  state = {
    isMenuOpen: false,
    generalFilterValue: '',
  };
  closeMenu() {
    this.setState({ isMenuOpen: false });
  }

  handleGeneralSearch = debounce((filter) => {
    if (filter && filter.length > 2) {
      this.props.startGeneralSearch(filter);
    }
  }, MILLISECONDS_TO_WAIT_REQUEST_AUTOCOMPLETE);

  cleanGeneralSearch() {
    this.setState({ generalFilterValue: '' });
    this.props.cleanGeneralSearch();
  }

  renderBrand() {
    return (
      <Navbar.Brand className="top-header__brand" as={Link} to="/">
        <div className="top-header__brand__logo-icon"></div>
      </Navbar.Brand>
    );
  }
  renderHeaderTopMenu() {
    const { t } = this.props;
    return (
      <Nav className="top-header__nav-paths mx-auto">
        <Nav.Link as={NavLink} activeClassName="active" to="/dashboard">
          {t('header.main-menu.home')}
        </Nav.Link>
        {this.props.view_community && (
          <Nav.Link as={NavLink} activeClassName="active" to="/community">
            {t('header.main-menu.community')}
          </Nav.Link>
        )}
        {this.props.view_network && (
          <Nav.Link as={NavLink} activeClassName="active" to="/network">
            {t('header.main-menu.network')}
          </Nav.Link>
        )}
        {/* <Nav.Link as={NavLink} activeClassName="active" to="/contracts">
                    {t("header.main-menu.contracts")}
                </Nav.Link> */}
      </Nav>
    );
  }

  renderSearchItem() {
    return (
      <Nav.Item>
        <Form inline>
          <Form.Group className="search-global">
            <Form.Control
              type="text"
              placeholder="Search"
              value={this.state.generalFilterValue}
              onChange={(event) => {
                this.setState({ generalFilterValue: event.target.value });
                this.handleGeneralSearch(event.target.value);
              }}
            />
            {this.state.generalFilterValue && (
              <div
                className="clear-input-cta"
                onClick={() => {
                  this.cleanGeneralSearch();
                }}
              >
                <FontAwesomeIcon className="clear-input" icon={faTimes} />
              </div>
            )}
          </Form.Group>
        </Form>
      </Nav.Item>
    );
  }
  renderNotificationsItem() {
    return (
      <Nav.Item>
        <Dropdown>
          <Dropdown.Toggle as="span">
            <i className="icon-notification"></i>
          </Dropdown.Toggle>
          <Dropdown.Menu className="dropdown-menu-right">
            <Dropdown.Item href="#/notification-1">Notification 1</Dropdown.Item>
            <Dropdown.Item href="#/notification-2">Notification 2</Dropdown.Item>
            <Dropdown.Item href="#/notification-3">Notification 3</Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Nav.Item>
    );
  }

  renderMoreItem() {
    return (
      <Nav.Item>
        <Dropdown>
          <Dropdown.Toggle as="span">
            <i className="icon-more"></i>
          </Dropdown.Toggle>
          <Dropdown.Menu className="dropdown-menu-right">
            <Dropdown.Item href="#/notification-1">Notification 1</Dropdown.Item>
            <Dropdown.Item href="#/notification-2">Notification 2</Dropdown.Item>
            <Dropdown.Item href="#/notification-3">Notification 3</Dropdown.Item>
          </Dropdown.Menu>
        </Dropdown>
      </Nav.Item>
    );
  }

  renderLogoutItem() {
    return (
      <Nav.Item className="top-header__user-block__logout">
        <Logout />
      </Nav.Item>
    );
  }

  renderProfileSettingsItem() {
    return (
      <Nav.Item className="top-header__user-block__profile px-0">
        <Link to="/personal-area/profile-settings">
          <div className="top-header__user-block__profile__icon"></div>
        </Link>
      </Nav.Item>
    );
  }

  renderMobileMenu() {
    return (
      <Navbar className="top-header">
        {this.renderBrand()}
        {this.renderSearchItem()}
      </Navbar>
    );
  }

  renderMobileNavigator() {
    const { t } = this.props;

    const toggleMenuData = [
      {
        header: t('header.main-menu.community'),
        subOptions: [
          {
            i18nText: t(`main-entity-name/organizations`),
            path: '/community/organizations',
            iconClass: 'icon-organization',
          },
          { i18nText: t(`main-entity-name/contacts`), path: '/community/contacts', iconClass: 'icon-contact' },
          { i18nText: t(`main-entity-name/groups`), path: '/community/groups', iconClass: 'icon-groups' },
        ],
      },
      {
        header: t('header.main-menu.network'),
        subOptions: [
          {
            i18nText: 'Organizations',
            path: '/network/customers',
            iconClass: 'icon-organization',
            subSubOptions: [
              { i18nText: t('main-entity-name/customers'), path: '/network/customers' },
              { i18nText: t('main-entity-name/end-users'), path: '/network/end-users' },
              { i18nText: t('main-entity-name/providers'), path: '/network/providers' },
              { i18nText: t('main-entity-name/site-owners'), path: '/network/site-owners' },
            ],
          },
          {
            i18nText: t('section/network/equipment-cables'),
            path: '/network/cables',
            iconClass: 'icon-organization',
            subSubOptions: [
              { i18nText: t('main-entity-name/ports'), path: '/network/ports' },
              { i18nText: t('main-entity-name/cables'), path: '/network/cables' },
              { i18nText: t('main-entity-name/hosts'), path: '/network/hosts' },
              { i18nText: t('main-entity-name/firewalls'), path: '/network/firewalls' },
              { i18nText: t('main-entity-name/routers'), path: '/network/routers' },
              { i18nText: t('main-entity-name/switches'), path: '/network/switches' },
              { i18nText: t('main-entity-name/external-equipment'), path: '/network/external-equipments' },
              { i18nText: t('main-entity-name/optical-nodes'), path: '/network/optical-nodes' },
              { i18nText: t('main-entity-name/odfs'), path: '/network/odfs' },
            ],
          },
          {
            i18nText: t('section/network/optical-layers'),
            path: '/network/optical-nodes',
            iconClass: 'icon-organization',
            subSubOptions: [
              { i18nText: t('main-entity-name/optical-filters'), path: 'optical-filters' },
              { i18nText: t('main-entity-name/optical-links'), path: 'optical-links' },
              { i18nText: t('main-entity-name/optical-multiplex-sections'), path: 'optical-multiplex-sections' },
              { i18nText: t('main-entity-name/optical-paths'), path: 'optical-paths' },
            ],
          },
          {
            i18nText: t('section/network/peering'),
            path: '/network/peering-partners',
            iconClass: 'icon-organization',
            subSubOptions: [
              { i18nText: t('main-entity-name/peering-groups'), path: '/network/peering-groups' },
              { i18nText: t('main-entity-name/peering-partners'), path: '/network/peering-partners' },
            ],
          },
          {
            i18nText: t('section/network/location'),
            path: '/network/location-sites',
            iconClass: 'icon-organization',
            subSubOptions: [
              { i18nText: t('main-entity-name/location-racks'), path: '/network/location-racks' },
              { i18nText: t('main-entity-name/location-rooms'), path: '/network/location-rooms' },
              { i18nText: t('main-entity-name/location-sites'), path: '/network/location-sites' },
            ],
          },
          {
            i18nText: t('main-entity-name/services'),
            path: formattedServicesData[0].path,
            iconClass: 'icon-organization',
            subSubOptions: formattedServicesData,
          },
        ],
      },
    ];

    return (
      <AccordionMenuOptions data={toggleMenuData} onSelectedOption={() => this.closeMenu()}></AccordionMenuOptions>
    );
  }

  renderPanelInside() {
    return (
      <main id="page-wrap">
        <Navbar className="top-header">
          <Nav className="top-header__user-block">
            {this.renderProfileSettingsItem()}
            {/* {this.renderNotificationsItem()} */}
            {/* {this.renderMoreItem()} */}
            {this.renderLogoutItem()}
          </Nav>
        </Navbar>
        {this.renderMobileNavigator()}
      </main>
    );
  }

  renderMobileLateralPanelMenu() {
    return (
      <BurgerMenu
        right
        disableAutoFocus
        disableOverlayClick={() => this.closeMenu()}
        customCrossIcon={<div className="row-cross-remove-cta"></div>}
        isOpen={this.state.isMenuOpen}
        onStateChange={(newState) => {
          this.setState({ isMenuOpen: newState.isOpen });
        }}
        outerContainerId={'outer-container'}
        pageWrapId={'page-wrap'}
      >
        {this.renderPanelInside()}
      </BurgerMenu>
    );
  }

  renderMobileHeader() {
    return (
      <header id="outer-container">
        <Navbar>
          {this.renderMobileMenu()}
          {/* {this.renderDesktopHeader()} */}
          {/* <Navbar className="top-header__nav">
                      <Nav className="top-header__nav__brand">{this.renderBrand()}</Nav>
                  </Navbar>
                  <Navbar className="top-header__nav">
                      <Nav className="top-header__nav__menu desktop-only">{this.renderHeaderTopMenu()}</Nav>
                  </Navbar>
                  <Navbar className="top-header__nav">
                      <Nav className="top-header__nav__right-block desktop-only">
                          {this.renderSearch()}
                          {this.renderNotifications()}
                          {this.renderMore()}
                          {this.renderProfileSettings()}
                          {this.renderLogout()}
                      </Nav>
                  </Navbar>
                  <Navbar className="top-header__nav">
                      <Nav className="top-header__nav__burger-button mobile-tablet-only">
                          <BurgerButton></BurgerButton>
                      </Nav>
                  </Navbar> */}
        </Navbar>
        {this.renderMobileLateralPanelMenu()}
      </header>
    );
  }

  renderDesktopHeader() {
    return (
      <Navbar className="top-header">
        {this.renderBrand()}
        {this.renderHeaderTopMenu()}
        <Nav className="top-header__user-block">
          {this.renderSearchItem()}
          {/* {this.renderNotificationsItem()} */}
          {/* {this.renderMoreItem()} */}
          {this.renderProfileSettingsItem()}
          {this.renderLogoutItem()}
        </Nav>
      </Navbar>
    );
  }

  render() {
    return (
      <div>
        <MobileView>{this.renderMobileHeader()}</MobileView>
        <BrowserView>{this.renderDesktopHeader()}</BrowserView>
      </div>
    );
  }
}

TopHeader.propTypes = {
  t: PropTypes.func.isRequired,
  view_community: PropTypes.bool,
  view_services: PropTypes.bool,
  view_network: PropTypes.bool,
  landing_page: PropTypes.string,
};

export default withTranslation()(TopHeader);
