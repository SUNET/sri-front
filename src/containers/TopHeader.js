import { connect } from 'react-redux';

import * as generalSearchActions from '../actions/GeneralSearch';
import TopHeader from '../components/TopHeader';

const mapStateToProps = (state, props) => {
  return {
    view_community: state.app.user.view_community,
    view_services: state.app.user.view_services,
    view_network: state.app.user.view_network,
    landing_page: state.app.user.landing_page,
    generalFilter: state.generalSearch.filter,
  };
};

const mapDispatchToProps = (dispatch, props) => {
  return {
    startGeneralSearch: (filterText) => {
      dispatch(generalSearchActions.startGeneralSearch(filterText));
    },
    cleanGeneralSearch: () => {
      dispatch(generalSearchActions.cleanGeneralSearch());
    },
  };
};

const TopHeaderContainer = connect(mapStateToProps, mapDispatchToProps)(TopHeader);

export default TopHeaderContainer;
