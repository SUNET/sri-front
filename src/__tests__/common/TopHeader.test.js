import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { TopHeader } from '../../components/TopHeader';

const mockStore = configureMockStore();

let TopHeaderWrapper;
let TopHeaderComponent;

const defaultProps = {
  t: jest.fn((t) => t),
  view_community: true,
  view_services: true,
  view_network: true,
  landing_page: 'community',
};

const historyMock = {
  push: jest.fn(),
  goBack: jest.fn(),
  location: {
    pathname: '/dashboard',
  },
  createHref: jest.fn(),
  listen: () => {},
};

const CreateTopHeaderWrapper = (props) => {
  const allProps = { ...defaultProps, ...props };
  const store = mockStore({
    view_community: true,
    view_services: true,
    view_network: true,
    landing_page: true,
  });
  return (
    <Provider history={historyMock} store={store}>
      <Router history={historyMock}>
        <TopHeader {...allProps} />
      </Router>
    </Provider>
  );
};

const mountTopHeaderWrapper = (props) => mount(<CreateTopHeaderWrapper {...props} />);

beforeEach(() => {
  jest.clearAllMocks();
  TopHeaderWrapper = mountTopHeaderWrapper();
  TopHeaderComponent = TopHeaderWrapper.find('TopHeader');
});
afterEach(() => {
  TopHeaderWrapper.unmount();
});

describe('Some Component', () => {
  it('Renders correctly', () => {
    expect(toJson(TopHeaderWrapper)).toMatchSnapshot();
    expect(TopHeaderComponent.find('nav.top-header')).toHaveLength(1);
  });
});
describe('Elements', () => {
  it('Brand', () => {
    const brand = TopHeaderComponent.find('a.top-header__brand');
    const logoContainer = brand.find('.top-header__brand__logo-icon');
    expect(brand).toHaveLength(1);
    expect(logoContainer).toHaveLength(1);
  });
  it('Navbar', () => {
    const navBarPaths = TopHeaderComponent.find('div.top-header__nav-paths');
    const links = navBarPaths.find('a');
    const homeLink = links.find('[children="header.main-menu.home"]');
    const communityLink = links.find('[children="header.main-menu.community"]');
    const networkLink = links.find('[children="header.main-menu.network"]');
    expect(navBarPaths).toHaveLength(1);
    expect(links).toHaveLength(3);
    expect(homeLink).toHaveLength(1);
    expect(communityLink).toHaveLength(1);
    expect(networkLink).toHaveLength(1);
    expect(homeLink.prop('className').match('active')).toHaveLength(1);
  });
  it('Persona area', () => {
    const editProfileButton = TopHeaderComponent.find('div.top-header__user-block__profile');
    const link = editProfileButton.find('a');
    const linkIcon = link.find('.top-header__user-block__profile__icon');
    expect(editProfileButton).toHaveLength(1);
    expect(link).toHaveLength(1);
    expect(linkIcon).toHaveLength(1);
  });
  it('Logout Button', () => {
    const logoutWrapper = TopHeaderComponent.find('div.top-header__user-block__logout');
    expect(logoutWrapper).toHaveLength(1);
    expect(logoutWrapper.find('Logout')).toHaveLength(1);
  });
});
