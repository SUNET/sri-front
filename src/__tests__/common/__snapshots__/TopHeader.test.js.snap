// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Some Component Renders correctly 1`] = `
<CreateTopHeaderWrapper>
  <Provider
    history={
      Object {
        "createHref": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": null,
              },
            ],
            Array [
              Object {
                "hash": "",
                "pathname": "/dashboard",
                "search": "",
                "state": null,
              },
            ],
            Array [
              Object {
                "hash": "",
                "pathname": "/community",
                "search": "",
                "state": null,
              },
            ],
            Array [
              Object {
                "hash": "",
                "pathname": "/network",
                "search": "",
                "state": null,
              },
            ],
            Array [
              Object {
                "hash": "",
                "pathname": "/personal-area/profile-settings",
                "search": "",
                "state": null,
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "goBack": [MockFunction],
        "listen": [Function],
        "location": Object {
          "pathname": "/dashboard",
        },
        "push": [MockFunction],
      }
    }
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Router
      history={
        Object {
          "createHref": [MockFunction] {
            "calls": Array [
              Array [
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": null,
                },
              ],
              Array [
                Object {
                  "hash": "",
                  "pathname": "/dashboard",
                  "search": "",
                  "state": null,
                },
              ],
              Array [
                Object {
                  "hash": "",
                  "pathname": "/community",
                  "search": "",
                  "state": null,
                },
              ],
              Array [
                Object {
                  "hash": "",
                  "pathname": "/network",
                  "search": "",
                  "state": null,
                },
              ],
              Array [
                Object {
                  "hash": "",
                  "pathname": "/personal-area/profile-settings",
                  "search": "",
                  "state": null,
                },
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
              Object {
                "type": "return",
                "value": undefined,
              },
            ],
          },
          "goBack": [MockFunction],
          "listen": [Function],
          "location": Object {
            "pathname": "/dashboard",
          },
          "push": [MockFunction],
        }
      }
    >
      <TopHeader
        landing_page="community"
        t={
          [MockFunction] {
            "calls": Array [
              Array [
                "header.main-menu.community",
              ],
              Array [
                "main-entity-name/organizations",
              ],
              Array [
                "main-entity-name/contacts",
              ],
              Array [
                "main-entity-name/groups",
              ],
              Array [
                "header.main-menu.network",
              ],
              Array [
                "main-entity-name/customers",
              ],
              Array [
                "main-entity-name/end-users",
              ],
              Array [
                "main-entity-name/providers",
              ],
              Array [
                "main-entity-name/site-owners",
              ],
              Array [
                "section/network/equipment-cables",
              ],
              Array [
                "main-entity-name/ports",
              ],
              Array [
                "main-entity-name/cables",
              ],
              Array [
                "main-entity-name/hosts",
              ],
              Array [
                "main-entity-name/firewalls",
              ],
              Array [
                "main-entity-name/routers",
              ],
              Array [
                "main-entity-name/switches",
              ],
              Array [
                "main-entity-name/external-equipment",
              ],
              Array [
                "main-entity-name/optical-nodes",
              ],
              Array [
                "main-entity-name/odfs",
              ],
              Array [
                "section/network/optical-layers",
              ],
              Array [
                "main-entity-name/optical-filters",
              ],
              Array [
                "main-entity-name/optical-links",
              ],
              Array [
                "main-entity-name/optical-multiplex-sections",
              ],
              Array [
                "main-entity-name/optical-paths",
              ],
              Array [
                "section/network/peering",
              ],
              Array [
                "main-entity-name/peering-groups",
              ],
              Array [
                "main-entity-name/peering-partners",
              ],
              Array [
                "section/network/location",
              ],
              Array [
                "main-entity-name/location-racks",
              ],
              Array [
                "main-entity-name/location-rooms",
              ],
              Array [
                "main-entity-name/location-sites",
              ],
              Array [
                "header.main-menu.home",
              ],
              Array [
                "header.main-menu.community",
              ],
              Array [
                "header.main-menu.network",
              ],
            ],
            "results": Array [
              Object {
                "type": "return",
                "value": "header.main-menu.community",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/organizations",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/contacts",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/groups",
              },
              Object {
                "type": "return",
                "value": "header.main-menu.network",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/customers",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/end-users",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/providers",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/site-owners",
              },
              Object {
                "type": "return",
                "value": "section/network/equipment-cables",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/ports",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/cables",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/hosts",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/firewalls",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/routers",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/switches",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/external-equipment",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/optical-nodes",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/odfs",
              },
              Object {
                "type": "return",
                "value": "section/network/optical-layers",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/optical-filters",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/optical-links",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/optical-multiplex-sections",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/optical-paths",
              },
              Object {
                "type": "return",
                "value": "section/network/peering",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/peering-groups",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/peering-partners",
              },
              Object {
                "type": "return",
                "value": "section/network/location",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/location-racks",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/location-rooms",
              },
              Object {
                "type": "return",
                "value": "main-entity-name/location-sites",
              },
              Object {
                "type": "return",
                "value": "header.main-menu.home",
              },
              Object {
                "type": "return",
                "value": "header.main-menu.community",
              },
              Object {
                "type": "return",
                "value": "header.main-menu.network",
              },
            ],
          }
        }
        view_community={true}
        view_network={true}
        view_services={true}
      >
        <div>
          <MobileView />
          <BrowserView>
            <div>
              <Navbar
                className="top-header"
                collapseOnSelect={false}
                expand={true}
                variant="light"
              >
                <nav
                  className="top-header navbar navbar-expand navbar-light"
                >
                  <NavbarBrand
                    as={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "displayName": "Link",
                        "propTypes": Object {
                          "innerRef": [Function],
                          "onClick": [Function],
                          "replace": [Function],
                          "target": [Function],
                          "to": [Function],
                        },
                        "render": [Function],
                      }
                    }
                    className="top-header__brand"
                    to="/"
                  >
                    <Link
                      className="top-header__brand navbar-brand"
                      to="/"
                    >
                      <LinkAnchor
                        className="top-header__brand navbar-brand"
                        navigate={[Function]}
                      >
                        <a
                          className="top-header__brand navbar-brand"
                          onClick={[Function]}
                        >
                          <div
                            className="top-header__brand__logo-icon"
                          />
                        </a>
                      </LinkAnchor>
                    </Link>
                  </NavbarBrand>
                  <Nav
                    className="top-header__nav-paths mx-auto"
                    fill={false}
                    justify={false}
                  >
                    <ForwardRef
                      as="div"
                      className="top-header__nav-paths mx-auto navbar-nav"
                      onSelect={[Function]}
                    >
                      <div
                        className="top-header__nav-paths mx-auto navbar-nav"
                        onKeyDown={[Function]}
                      >
                        <NavLink
                          activeClassName="active"
                          as={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "displayName": "NavLink",
                              "propTypes": Object {
                                "activeClassName": [Function],
                                "activeStyle": [Function],
                                "aria-current": [Function],
                                "className": [Function],
                                "exact": [Function],
                                "innerRef": [Function],
                                "isActive": [Function],
                                "location": [Function],
                                "onClick": [Function],
                                "replace": [Function],
                                "sensitive": [Function],
                                "strict": [Function],
                                "style": [Function],
                                "target": [Function],
                                "to": [Function],
                              },
                              "render": [Function],
                            }
                          }
                          disabled={false}
                          to="/dashboard"
                        >
                          <ForwardRef
                            activeClassName="active"
                            as={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "displayName": "NavLink",
                                "propTypes": Object {
                                  "activeClassName": [Function],
                                  "activeStyle": [Function],
                                  "aria-current": [Function],
                                  "className": [Function],
                                  "exact": [Function],
                                  "innerRef": [Function],
                                  "isActive": [Function],
                                  "location": [Function],
                                  "onClick": [Function],
                                  "replace": [Function],
                                  "sensitive": [Function],
                                  "strict": [Function],
                                  "style": [Function],
                                  "target": [Function],
                                  "to": [Function],
                                },
                                "render": [Function],
                              }
                            }
                            className="nav-link"
                            disabled={false}
                            to="/dashboard"
                          >
                            <NavLink
                              activeClassName="active"
                              className="nav-link"
                              data-rb-event-key={null}
                              disabled={false}
                              onClick={[Function]}
                              to="/dashboard"
                            >
                              <Link
                                aria-current="page"
                                className="nav-link active"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                                style={Object {}}
                                to={
                                  Object {
                                    "hash": "",
                                    "pathname": "/dashboard",
                                    "search": "",
                                    "state": null,
                                  }
                                }
                              >
                                <LinkAnchor
                                  aria-current="page"
                                  className="nav-link active"
                                  data-rb-event-key={null}
                                  disabled={false}
                                  navigate={[Function]}
                                  onClick={[Function]}
                                  style={Object {}}
                                >
                                  <a
                                    aria-current="page"
                                    className="nav-link active"
                                    data-rb-event-key={null}
                                    disabled={false}
                                    onClick={[Function]}
                                    style={Object {}}
                                  >
                                    header.main-menu.home
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </NavLink>
                          </ForwardRef>
                        </NavLink>
                        <NavLink
                          activeClassName="active"
                          as={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "displayName": "NavLink",
                              "propTypes": Object {
                                "activeClassName": [Function],
                                "activeStyle": [Function],
                                "aria-current": [Function],
                                "className": [Function],
                                "exact": [Function],
                                "innerRef": [Function],
                                "isActive": [Function],
                                "location": [Function],
                                "onClick": [Function],
                                "replace": [Function],
                                "sensitive": [Function],
                                "strict": [Function],
                                "style": [Function],
                                "target": [Function],
                                "to": [Function],
                              },
                              "render": [Function],
                            }
                          }
                          disabled={false}
                          to="/community"
                        >
                          <ForwardRef
                            activeClassName="active"
                            as={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "displayName": "NavLink",
                                "propTypes": Object {
                                  "activeClassName": [Function],
                                  "activeStyle": [Function],
                                  "aria-current": [Function],
                                  "className": [Function],
                                  "exact": [Function],
                                  "innerRef": [Function],
                                  "isActive": [Function],
                                  "location": [Function],
                                  "onClick": [Function],
                                  "replace": [Function],
                                  "sensitive": [Function],
                                  "strict": [Function],
                                  "style": [Function],
                                  "target": [Function],
                                  "to": [Function],
                                },
                                "render": [Function],
                              }
                            }
                            className="nav-link"
                            disabled={false}
                            to="/community"
                          >
                            <NavLink
                              activeClassName="active"
                              className="nav-link"
                              data-rb-event-key={null}
                              disabled={false}
                              onClick={[Function]}
                              to="/community"
                            >
                              <Link
                                aria-current={null}
                                className="nav-link"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                                to={
                                  Object {
                                    "hash": "",
                                    "pathname": "/community",
                                    "search": "",
                                    "state": null,
                                  }
                                }
                              >
                                <LinkAnchor
                                  aria-current={null}
                                  className="nav-link"
                                  data-rb-event-key={null}
                                  disabled={false}
                                  navigate={[Function]}
                                  onClick={[Function]}
                                >
                                  <a
                                    aria-current={null}
                                    className="nav-link"
                                    data-rb-event-key={null}
                                    disabled={false}
                                    onClick={[Function]}
                                  >
                                    header.main-menu.community
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </NavLink>
                          </ForwardRef>
                        </NavLink>
                        <NavLink
                          activeClassName="active"
                          as={
                            Object {
                              "$$typeof": Symbol(react.forward_ref),
                              "displayName": "NavLink",
                              "propTypes": Object {
                                "activeClassName": [Function],
                                "activeStyle": [Function],
                                "aria-current": [Function],
                                "className": [Function],
                                "exact": [Function],
                                "innerRef": [Function],
                                "isActive": [Function],
                                "location": [Function],
                                "onClick": [Function],
                                "replace": [Function],
                                "sensitive": [Function],
                                "strict": [Function],
                                "style": [Function],
                                "target": [Function],
                                "to": [Function],
                              },
                              "render": [Function],
                            }
                          }
                          disabled={false}
                          to="/network"
                        >
                          <ForwardRef
                            activeClassName="active"
                            as={
                              Object {
                                "$$typeof": Symbol(react.forward_ref),
                                "displayName": "NavLink",
                                "propTypes": Object {
                                  "activeClassName": [Function],
                                  "activeStyle": [Function],
                                  "aria-current": [Function],
                                  "className": [Function],
                                  "exact": [Function],
                                  "innerRef": [Function],
                                  "isActive": [Function],
                                  "location": [Function],
                                  "onClick": [Function],
                                  "replace": [Function],
                                  "sensitive": [Function],
                                  "strict": [Function],
                                  "style": [Function],
                                  "target": [Function],
                                  "to": [Function],
                                },
                                "render": [Function],
                              }
                            }
                            className="nav-link"
                            disabled={false}
                            to="/network"
                          >
                            <NavLink
                              activeClassName="active"
                              className="nav-link"
                              data-rb-event-key={null}
                              disabled={false}
                              onClick={[Function]}
                              to="/network"
                            >
                              <Link
                                aria-current={null}
                                className="nav-link"
                                data-rb-event-key={null}
                                disabled={false}
                                onClick={[Function]}
                                to={
                                  Object {
                                    "hash": "",
                                    "pathname": "/network",
                                    "search": "",
                                    "state": null,
                                  }
                                }
                              >
                                <LinkAnchor
                                  aria-current={null}
                                  className="nav-link"
                                  data-rb-event-key={null}
                                  disabled={false}
                                  navigate={[Function]}
                                  onClick={[Function]}
                                >
                                  <a
                                    aria-current={null}
                                    className="nav-link"
                                    data-rb-event-key={null}
                                    disabled={false}
                                    onClick={[Function]}
                                  >
                                    header.main-menu.network
                                  </a>
                                </LinkAnchor>
                              </Link>
                            </NavLink>
                          </ForwardRef>
                        </NavLink>
                      </div>
                    </ForwardRef>
                  </Nav>
                  <Nav
                    className="top-header__user-block"
                    fill={false}
                    justify={false}
                  >
                    <ForwardRef
                      as="div"
                      className="top-header__user-block navbar-nav"
                      onSelect={[Function]}
                    >
                      <div
                        className="top-header__user-block navbar-nav"
                        onKeyDown={[Function]}
                      >
                        <NavItem>
                          <div
                            className="nav-item"
                          >
                            <Form
                              inline={true}
                            >
                              <form
                                className="form-inline"
                              >
                                <FormGroup
                                  className="search-global"
                                >
                                  <div
                                    className="search-global form-group"
                                  >
                                    <FormControl
                                      onChange={[Function]}
                                      placeholder="Search"
                                      type="text"
                                      value=""
                                    >
                                      <input
                                        className="form-control"
                                        onChange={[Function]}
                                        placeholder="Search"
                                        type="text"
                                        value=""
                                      />
                                    </FormControl>
                                  </div>
                                </FormGroup>
                              </form>
                            </Form>
                          </div>
                        </NavItem>
                        <NavItem
                          className="top-header__user-block__profile px-0"
                        >
                          <div
                            className="top-header__user-block__profile px-0 nav-item"
                          >
                            <Link
                              to="/personal-area/profile-settings"
                            >
                              <LinkAnchor
                                navigate={[Function]}
                              >
                                <a
                                  onClick={[Function]}
                                >
                                  <div
                                    className="top-header__user-block__profile__icon"
                                  />
                                </a>
                              </LinkAnchor>
                            </Link>
                          </div>
                        </NavItem>
                        <NavItem
                          className="top-header__user-block__logout"
                        >
                          <div
                            className="top-header__user-block__logout nav-item"
                          >
                            <Logout>
                              <i
                                className="icon-logout"
                                onClick={[Function]}
                                title="Logout"
                              />
                            </Logout>
                          </div>
                        </NavItem>
                      </div>
                    </ForwardRef>
                  </Nav>
                </nav>
              </Navbar>
            </div>
          </BrowserView>
        </div>
      </TopHeader>
    </Router>
  </Provider>
</CreateTopHeaderWrapper>
`;
